### PR TITLE
Preliminary SeqBits, TriggeredMoves, NetFx. PowerStance. More.

### LINKED ISSUE #188: MapInstance::on_change_stance is unhandled
### Summary
This is one of many events in MapInstance.cpp that is unhandled in one way or another.
### Details
Begins on line 666 of MapInstance.cpp.
When called, it gives "Unhandled change stance request" to qWarning(). It also has logic to differentiate between entering and exiting a stance, though it only gives debug messages.

### PR DESCRIPTION
**Pull request by broxen**
Most of this was found in my Movement branch, and I'm pulling it out to merge here because it has low impact on existing functionality.

Closes #188 

In particular, most of this doesn't change any functional behavior with the exception of PowerStances which now work as expect (fire power at target, and player goes into fight stance)

Additions/modifications proposed in this pull request:
- Preliminary Support for SeqBits, TriggeredMoves, NetFx
- SendStance and ChangeStance fixed and tested.
- Preliminary Movement.cpp/h
- Support for lastOnline, which now should be fired during character updates
- NPC Factions have been fixed
- Toggle movement authority with `/movement` -- Wont work until we disable the work-around from Nem, but needed for the rest of my movement patch coming down the pipeline.
- Lots of variables renamed and code cleanups informed by my explorations into Movement Issues. Example: `m_extra_info` now becomes `m_has_interp` to better reflect it's purpose.

### CODE REVIEW

File: Projects/CoX/Common/GameData/seq_definitions.h [hunk 1]
```diff
@@ -20,386 +20,391 @@ struct AnimTrack;
...
-    SEVEN=365,
-    EIGHT=366,
-    NINE=367,
-    TEN=368,
-    ZERO=369,
 
-    NON_EXISTING=~0U
+enum class SeqBitNames : uint32_t
+{
+    SB_INVALID_BIT=0,
```
[the remark this answers is not in the document]
Comment (broxen) [reply Projects/CoX/Common/GameData/seq_definitions.h]: TIL. Done

File: Projects/CoX/Servers/MapServer/EntityUpdateCodec.cpp [hunk 2]
```diff
@@ -162,94 +161,86 @@ void storePosUpdate(const Entity &src, bool just_created, BitStream &bs)
...
     }
 }
-void sendSeqTriggeredMoves(const Entity &/*src*/,BitStream &bs)
+void sendSeqTriggeredMoves(const Entity &src,BitStream &bs)
 {
     PUTDEBUG("before sendSeqTriggeredMoves");
-    uint32_t num_moves = 0; // FixMe: num_moves is never modified and the body of the for loop below will never fire.
-    qCDebug(logAnimations, "Sending seq triggered moves %d", num_moves);
+    qCDebug(logAnimations, "Sending seq triggered moves %d", src.m_triggered_moves.size());
+    if(src.m_triggered_moves.size() < 20)
```
[the remark this answers is not in the document]
Comment (broxen) [reply Projects/CoX/Servers/MapServer/EntityUpdateCodec.cpp]: Only that client stops counting queued moves after 20. I'll adjust to a comment, as it doesn't appear to "require" 20

File: Projects/CoX/Common/NetStructures/CharacterHelpers.cpp [hunk 3]
```diff
@@ -101,6 +103,11 @@ void setInf(Character &c, uint32_t val) { c.m_char_data.m_influence = val; }
 void setDescription(Character &c, QString val) { c.m_char_data.m_character_description = val; }
 void setBattleCry(Character &c, QString val) { c.m_char_data.m_battle_cry = val; }
 
+void setLastOnline(Character &c)
```
Comment (nemerle) [inline Projects/CoX/Common/NetStructures/CharacterHelpers.cpp]: this does not allow anyone to `setLastOnline` but it's a pretty good way of `setLastOnlineToNow`, or maybe a plain `updateLastOnlineTime` ? :D

File: Projects/CoX/Common/NetStructures/Movement.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+{
+    e.m_entity_data.m_pos = pos;
+    e.m_full_update_count = 10;
+}
+
+// Move to Sequences or Triggers files later
+void addTriggeredMove(Entity &e, TriggeredMove &trig)
+{
+    e.m_rare_update = true;
+    e.m_triggered_moves.at(trig.m_move_idx) = trig;
```
Comment (nemerle) [inline Projects/CoX/Common/NetStructures/Movement.cpp]: This is a bit strange, this is more of an `set` then an `add` :)

Comment (broxen) [reply Projects/CoX/Common/NetStructures/Movement.cpp]: Ah yes. This is all wrong now that triggeredmoves is a vector

File: Projects/CoX/Common/NetStructures/CharacterHelpers.cpp (hunk 3 above)
Comment (broxen) [reply Projects/CoX/Common/NetStructures/CharacterHelpers.cpp]: Agreed. Update sounds like the best verbiage

**Comment by broxen** [comment]
Barring any other feedback, I think this is ready for merge.

**Pull request merged**